tests/primer/primer_tool.py

```diff
@@ -255,14 +259,29 @@ def _create_comment(
                 comment += "\n</details>\n\n"
 
         if comment == "":
-            comment = "🤖 According to the primer, this change has **no effect** on the checked open source code. 🤖🎉"
+            comment = (
+                "🤖 According to the primer, this change has **no effect** on the"
+                " checked open source code. 🤖🎉\n\n"
+            )
         else:
             comment = (
-                "🤖 **Effect of this PR on checked open source code:** 🤖\n\n" + comment
+                f"🤖 **Effect of this PR on checked open source code:** 🤖\n\n{comment}"
             )
-
-        comment += f"*This comment was generated for commit {self.config.commit}*"
-
+        hash_information = (
+            f"*This comment was generated for commit {self.config.commit}*"
+        )
+        if len(comment) + len(hash_information) >= MAX_GITHUB_COMMENT_LENGTH:
+            truncation_information = (
+                f"*This comment was truncated because GitHub allows only"
+                f"{MAX_GITHUB_COMMENT_LENGTH} characters in a comment.*"
+            )
+            max_len = (
+                MAX_GITHUB_COMMENT_LENGTH
+                - len(hash_information)
+                - len(truncation_information)
+            )
+            comment = f"{comment[:max_len - 5]}...\n{truncation_information}"
```

Comment:
```suggestion
            comment = f"{comment[:max_len - 10]}...\n\n{truncation_information}\n\n"
```

Markdown needs two new lines to format nicely.